request/builder.js

```diff
@@ -0,0 +1,481 @@
+function TextEncoder() {
+}
+
+TextEncoder.prototype.encode = function (string) {
```

Comment:
This encode can be done by using third-party library ?

Reply:
I couldn't find a good text encoding library that is able to run on both of browser JS runtime and node.js JS runtime. (I guess it is because these two runtimes all have native ways doing that).

However, I think I could replace it with jsrsasign.
(Like what we did for base64 to hex, ... etc)

```
const jsrsasign = require('jsrsasign');
class TextEncoder {
  encode(string) {
    return !string
    ? new Uint8Array()
    : new Uint8Array(jsrsasign.hextoArrayBuffer(jsrsasign.utf8tohex(string)));
  }
}
```